Docs: Refactor "Environment variables" into 3 articles (Diátaxis) (#9966)
readthedocs/readthedocs.org: docs/user/guides/environment-variables.rst

How to use custom environment variables
=======================================

If extra environment variables are needed in the build process,
you can define them from the project's :term:`dashboard`.

.. seealso::

   :doc:`/environment-variables`
     Learn more about how Read the Docs applies environment variables in your builds.

Go to your project's :menuselection:`Admin --> Environment Variables` and click on :guilabel:`Add Environment Variable`.
You will then see the form for adding an environment variable:

.. image:: /img/screenshot_environment_variables.png
   :alt: Screenshot of the form for adding an environment variable

#. Fill in the :guilabel:`Name` field, this is the name of your variable, for instance ``SECRET_TOKEN`` or ``PIP_EXTRA_INDEX_URL``.
#. Fill in the :guilabel:`Value` field with the environment variable's value, for instance a secret token or a build configuration.
#. Check the :guilabel:`Public` option if you want to expose this environment variable
   to :doc:`builds from pull requests </pull-requests>`.

   .. warning::

>       If you mark this option, any user that can create a pull request

Reviewer: ```suggestion
      If you make your environment variable **public**, any user that can create a pull request
```

      on your repository will be able to see the value of this environment variable.
      In other words,

>       do not use this option if your environment variable is a secret.

Reviewer: ```suggestion
      **do not use this option if your environment variable is a secret.**
```

> .. note::

Reviewer: Don't love 2 callouts back to back, but seems OK in this case? Not sure this one needs a note vs. just normal text.

Author: It was fairly easy to convert it to normal text, yes :+1:

   Once you create an environment variable,
   you won't be able to edit or view its value in the :term:`dashboard`.

Finally, click on :guilabel:`Save`.
Your custom environment variable is immediately active for all future builds

> and you are ready to start using it.

Reviewer: 💯

> In the next sections,
> you can see a few examples of using environment variables. ⬇️

Reviewer: ```suggestion
Keep reading for a few examples of using environment variables. ⬇️
```

`next sections` is awkward.
